File: auth-center/auth-center-infra/src/main/java/com/jmsoftware/maf/authcenter/role/persistence/Role.java
package com.jmsoftware.maf.authcenter.role.persistence;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.jmsoftware.maf.springcloudstarter.database.BasePersistenceEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * <h1>Role</h1>
 * <p>
 * Role Persistence object class
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 6/27/2021 3:22 PM
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName(value = Role.TABLE_NAME)
public class Role extends BasePersistenceEntity {
    public static final String TABLE_NAME = "role";
    public static final String COL_NAME = "name";
    public static final String COL_DESCRIPTION = "description";
    /**
     * Role name
     */
    @TableField(value = COL_NAME)
    private String name;
    /**
     * Role description
     */
    @TableField(value = COL_DESCRIPTION)
    private String description;
}

File: auth-center/auth-center-infra/src/main/java/com/jmsoftware/maf/authcenter/role/persistence/Role.kt
package com.jmsoftware.maf.authcenter.role.persistence

import com.baomidou.mybatisplus.annotation.TableField
import com.baomidou.mybatisplus.annotation.TableName
import com.jmsoftware.maf.springcloudstarter.database.BasePersistenceEntity

/**
 * # Role
 *
 * Role Persistence object class
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/10/22 7:15 PM
 */
@TableName(value = Role.TABLE_NAME)
class Role(
    /**
     * Role name
     */
    @TableField(value = COL_NAME)
    val name: String,
    /**
     * Role description
     */
    @TableField(value = COL_DESCRIPTION)
    val description: String
) : BasePersistenceEntity() {
    companion object {
        const val TABLE_NAME = "role"
        const val COL_NAME = "name"
        const val COL_DESCRIPTION = "description"
    }
}
